core/src/main/java/com/graphhopper/routing/lm/LMAlgoFactoryDecorator.java

```diff
@@ -229,6 +229,11 @@ public RoutingAlgorithmFactory getDecoratedAlgorithmFactory(RoutingAlgorithmFact
         if (preparations.isEmpty())
             throw new IllegalStateException("No preparations added to this decorator");
 
+        // if no weighting or vehicle is specified for this request and there is only one preparation, use it
```

Comment:
This was very ugly: If the weighting was missing it was added to the map in `CH(!)AlgoFactoryDecorator` and `LMAlgoFactoryDecorator` relied on the CH decorator to add it before we got here...If we did not pay attention to the missing weighting here  (now that we no longer add the weighting in the CH decorator) we would get normal AStar instead of LM, see `GraphHopperIT#testFlexMode_631`.

Reply:
Uh, sounds ugly.